llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -442,46 +442,55 @@ def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack", [
      "generateScalarImplementation"]>,
   DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
 ]>{
-  let summary = "pack operation"; 
+  let summary = "pack operation";
   let description = [{
     The pack operation converts an `input` into a tiled and packed layout. The
     dimensions to be tiled are obtained from `dims_pos` and the size of the tile is
     obtained from `inner_tiles`. The dimensions listed in `dims_pos` do not need to
     be contiguous in which case the tile will get transposed.  We handle only full
-    tiles now; it is UB if the tile does not perfectly divide the dimension. We
-    will relax this constraint in the future when we add pad semantics to the
-    operation.
+    tiles if pad=false; it is UB if the tile does not perfectly divide the dimension..
+    If pad=true, it will pad along high dimensions, i.e., it pads at the bottom
+    and on the right if the input has rank 2, and the result type shape, will be
+    dynamic in any dimension if and only if the input shape is.
 
     Example KC_to_KCck:
-    
+
     ```mlir
-    iree_linalg_ext.pack %arg0 dims_pos = [1, 0] 
+    iree_linalg_ext.pack pad(false) %arg0 dims_pos = [1, 0]
       inner_tiles = [32, 8] into %arg1 : (memref<128x256xf32> memref<16x8x32x8xf32>)
     ```
-    
+
     Example NC_to_NCnc:
-    
+
     ```mlir
-    iree_linalg_ext.pack %arg0 dims_pos = [0, 1] 
+    iree_linalg_ext.pack pad(false) %arg0 dims_pos = [0, 1]
       inner_tiles = [8, 32] into %arg1 : (memref<128x256xf32> memref<16x8x8x32xf32>)
-    ``` 
-    
+    ```
+
     In both cases, dimension at position 0 in the input memref (128) is tiled
     with a factor of 8, while dimension at position 1 (256) is tiled with a factor
-    of 32. In the KC_to_KCck example, the point loops are interchanged. 
+    of 32. In the KC_to_KCck example, the point loops are interchanged.
+
+    Example NC_to_NCnc with padding:
+
+    ```mlir
+    iree_linalg_ext.pack pad(true) %arg dims_pos = [0, 1]
+      inner_tiles = [8, 2] into %arg1 : (memref<13x15xf32> memref<2x8x8x2xf32>)
+    ```
 
   }];
 
   let arguments = (ins Variadic<AnyShaped>:$inputs,
     Variadic<AnyShaped>:$outputs,
     DefaultValuedAttr<I64ArrayAttr, "{}">:$dims_pos,
     Variadic<Index>:$inner_tiles,
-    I64ArrayAttr:$static_inner_tiles);
+    I64ArrayAttr:$static_inner_tiles,
+    DefaultValuedAttr<BoolAttr, "false">:$pad);
```

Comment:
Instead of default `BoolAttr`, how about an optional `Value` operand for the pad value....?